### PR TITLE
Implementation of the KDE/photospline PSF  (9yr/11yr NT samples) as spatial PDF for flarestack

### PR DESCRIPTION
**Pull request by msackerm**
### Features
- defines NorthernTracksKDE(SignalSpatialPDF) class (name: "northern_tracks_kde")
    - class can be used with 3D (psi,sigma,logE) and 4D (psi, sigma,logE,gamma) spline fits of the PSF. 
    - can be used with all likelihoods, but psf does not change with gamma except in dedicated new likelihood classes
- defines two new Likelihood classes 
   - StandardKDEEnabledLLH(StandardLLH) (name: "standard_kde_enabled"), and 
   - StdMatrixKDEEnabledLLH(StandardOverlappingLLH) (name:"std_matrix_kde_enabled")
   - new llh classes support variations of the PSF with spectral index gamma

### Tests
- E^-2 PS sensitivity and discovery potential were calculated with 9yr and 11yr NT samples and agree with published values
- n_s and gamma bias have been checked and improvements found. They are consistent with tests done [here](https://user-web.icecube.wisc.edu/~tglauch/PS_Analysis/_build/html/performance.html).
- See also figures below.

image

image

### Other changes
- submitter.py was modified to address problems with DESY ht-condor. 
  - Submission command was adjusted to first switch to local directory of the job.submit file, since condor_submit would not accept a path prepending the file name. 
  - log dir creation was modified to create a new subdirectory for each cluster of jobs submitted (after crashing the DESY farm with 32000 log files in a directory).
-  nt_v005_p00.py was modified to use NTSeasonNewStyle (solves memory consumption issues)
- photospline (conda-forge) was added as a dependency for flarestack

### CODE REVIEW

**Comment by mlincett** [comment]
This is a very welcome addition.

I see the `photospline` dependency is handled entirely through conda, and `poetry` should work on top of that.

It could be nice if we could make photospline (and hence the KDE features) optional, but I am not sure whether there is an elegant way of achieving this (in the past I used to do `try import / except ImportError` and set a boolean switch).

I use `conda` for everything myself, but I am afraid it is somewhat an "expensive" requirement.

**Comment by JannisNe** [comment]
> This is a very welcome addition.
> 
> I see the `photospline` dependency is handled entirely through conda, and `poetry` should work on top of that.
> 
> It could be nice if we could make photospline (and hence the KDE features) optional, but I am not sure whether there is an elegant way of achieving this (in the past I used to do `try import / except ImportError` and set a boolean switch).
> 
> I use `conda` for everything myself, but I am afraid it is somewhat an "expensive" requirement.

@mlincett do you mean make it an optional dependency through `poetry`? That's not possible because `photospline` needs some pre-compiled things so it's only available through `conda`.
I guess the idea is to set up a conda environment with `conda create -f conda_env.yml` and then do the `poetry install` inside that environment. I will add some of thet to the README.

File: flarestack/analyses/benchmarks/benchmark_point_source_sensitivity_10yr.py
Comment (JannisNe) [inline flarestack/analyses/benchmarks/benchmark_point_source_sensitivity_10yr.py:1]: Are these changes necessary to make the script run with the newest `flarestack` version?

**Comment by mlincett** [comment]
> @mlincett do you mean make it an optional dependency through `poetry`? That's not possible because `photospline` needs some pre-compiled things so it's only available through `conda`.

I am aware of that. What I did in other circumstances is, in case of a missing "external" dependency, to catch the "ImportError" and add a gracious failure mode for when the downstream code tries to use the classes that require the dependency.

Yet this is not very elegant and can be a pain to maintain (especially in an heavily structured code like `flarestack`). If no one else is concerned with the conda / photospline dependency, I am fine with taking this step as well.

File: .github/workflows/continous_integration.yml [hunk 1]
```diff
@@ -33,27 +33,44 @@ jobs:
...
+            conda list
+            conda info -a
 
      # Setting up dependencies
       - name: Install dependencies
+        shell: bash -el {0}
         run: |
           python -m pip install --upgrade pip
           python -m pip install poetry
           python -m poetry install
```
Comment (mlincett) [inline .github/workflows/continous_integration.yml:62]: I think that installing `poetry` from/to within same virtual environment of the package can incidentally work but is  discouraged in general.

File: flarestack/analyses/benchmarks/benchmark_point_source_sensitivity_10yr.py
Comment (msackerm) [reply flarestack/analyses/benchmarks/benchmark_point_source_sensitivity_10yr.py:1]: Yes, partly. But there were also some unnecessary changes. I tried to clean it up.

**Comment by JannisNe** [comment]
Seems like there is some issue with `urllib3 v2.0.7` and python 3.10. I pinned v2.0.6 for now.
@mlincett good to merge?

**Review by mlincett — APPROVED** [review]
Looks good  to me!
